packages/webdriverio/src/scripts/isDisplayedInViewport.js

```diff
@@ -0,0 +1,35 @@
+
+/**
+ * check if element is visible and within the viewport
+ *
+ * @param {String}  elements  DOM elements to check against
+ *
+ * @see  waitForVisible
+ */
+
+export default function isDisplayedInViewport (elem) {
+    const dde = document.documentElement
+
+    let isWithinViewport = true
+    while (elem.parentNode && elem.parentNode.getBoundingClientRect) {
+        const elemDimension = elem.getBoundingClientRect()
+        const elemComputedStyle = window.getComputedStyle(elem)
+        const viewportDimension = {
+            width: dde.clientWidth,
+            height: dde.clientHeight
+        }
+
+        isWithinViewport = isWithinViewport &&
+                           (elemComputedStyle.display !== 'none' &&
```

Comment:
instead of using
`elemComputedStyle.display !== 'none' && elemComputedStyle.visibility === 'visible' && parseFloat(elemComputedStyle.opacity, 10) > 0`
I think it would make sense to use isVisible instead so you have two checks: isVisible and isWithinViewport :)

Reply:
theoretically yes but let's save a WebDriver request here and check these properties directly at the same time.